docs/rules/README.md

```diff
@@ -252,6 +252,8 @@ For example:
 | [vue/prefer-prop-type-boolean-first](./prefer-prop-type-boolean-first.md) | enforce `Boolean` comes first in component prop types | :bulb: | :warning: |
 | [vue/prefer-separate-static-class](./prefer-separate-static-class.md) | require static class names in template to be in a separate `class` attribute | :wrench: | :hammer: |
 | [vue/prefer-true-attribute-shorthand](./prefer-true-attribute-shorthand.md) | require shorthand form attribute when `v-bind` value is `true` | :bulb: | :hammer: |
+| [vue/prefer-type-emits-decl](./prefer-type-emits-decl.md) | enforce type-based `defineEmits` |  | :lipstick: |
+| [vue/prefer-type-props-decl](./prefer-type-props-decl.md) | enforce type-based `defineProps` |  | :lipstick: |
```

Comment:
I'd prefer to rename these rules to avoid the `decl` abbreviation and make it more clear what they do:

```suggestion
| [vue/prefer-type-only-define-emits](./prefer-type-emits-decl.md) | enforce type-based `defineEmits` |  | :lipstick: |
| [vue/prefer-type-only-define-props](./prefer-type-props-decl.md) | enforce type-based `defineProps` |  | :lipstick: |
```

Reply:
Sounds good, I will change this to `vue/prefer-type-based-define-emits` for consistency reasons.